mnefun/_mnefun.py

```diff
@@ -2405,24 +2411,55 @@ def do_preprocessing_combined(p, subjects, run_indices):
                              pick=True, skip_by_annotation='edge',
                              **fir_kwargs)
             events = fixed_len_events(p, raw)
+            rtmin = p.reject_tmin \
+                if p.reject_tmin is not None else p.tmin
+            rtmax = p.reject_tmax \
+                if p.reject_tmax is not None else p.tmax
             # do not mark eog channels bad
             meg, eeg = 'meg' in raw, 'eeg' in raw
             picks = pick_types(raw.info, meg=meg, eeg=eeg, eog=False,
                                exclude=[])
             assert p.auto_bad_flat is None or isinstance(p.auto_bad_flat, dict)
             assert p.auto_bad_reject is None or isinstance(p.auto_bad_reject,
-                                                           dict)
-            if p.auto_bad_reject is None and p.auto_bad_flat is None:
+                                                           dict) or \
+                   p.auto_bad_reject is 'auto'
+            if p.auto_bad_reject is 'auto':
+                print('    Auto bad channel selection active. '
+                      'Will try using Autoreject module to '
+                      'compute rejection criterion.\n' , end='')
+                try:
+                    from autoreject import get_rejection_threshold
+                    print('    Computing autoreject thresholds.\n', end='')
```

Comment:
Only put in `try` the minimal set of things expected to cause the error. So here do:
```
try:
    from autoreject import ...
except ImportError:
    print(...)
else:
    print(...)
    temp_epochs = ...
```
but actually if someone requests it but does not have it I would raise an error:
```
try:
   from autoreject import ...
except ImportError:
   raise ImportError(...)
print(...)
temp_epochs = ...
```